File: frontend/src/services/modules/auth.service.js
import { ENDPOINTS } from '../../api/endpoints'
import { request } from '../request/http'

export function register(payload) {
  return request.post(ENDPOINTS.auth.register, payload)
}

export function login(payload) {
  return request.post(ENDPOINTS.auth.login, payload)
}

export function fetchMe() {
  return request.get(ENDPOINTS.auth.me)
}

export function changePassword(payload) {
  return request.put(ENDPOINTS.auth.password, payload)
}

File: backend/demo/src/main/java/com/example/indras/auth/service/AuthService.java
package com.example.indras.auth.service;

import com.example.indras.auth.dto.ChangePasswordRequest;
import com.example.indras.auth.dto.LoginRequest;
import com.example.indras.auth.dto.RegisterRequest;
import com.example.indras.auth.vo.LoginResponseVO;
import com.example.indras.auth.vo.RegisterResponseVO;
import com.example.indras.auth.vo.UserVO;

public interface AuthService {

    RegisterResponseVO register(RegisterRequest request);

    LoginResponseVO login(LoginRequest request);

    UserVO currentUser(Long userId);

    boolean changePassword(Long userId, ChangePasswordRequest request);
}
